.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,156 @@
+    steps:
+      - uses: actions/checkout@v4
+      - run: mvn --non-recursive wrapper:wrapper -Dmaven=3.9.6
+      - name: Set up Java 11
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 11
+      - name: Performance Tests
+        run: ./mvnw install --projects performance-tests --also-make --activate-profiles all -DskipTests=true
+        env:
+          MAVEN_OPTS: "-Xmx1g"
+
+  checkstyle:
+    name: Checkstyle
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: mvn --non-recursive wrapper:wrapper -Dmaven=3.9.6
+      - name: Set up Java 11
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 11
+      - name: Checkstyle
+        run: ./mvnw install checkstyle:check --activate-profiles all -DskipTests=true
+        env:
+          MAVEN_OPTS: "-Xmx1g"
+
+  javadoc:
+    name: Javadoc 11
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: mvn --non-recursive wrapper:wrapper -Dmaven=3.9.6
+      - name: Set up Java 11
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 11
+      - name: JavaDoc Aggregate
+        run: ./mvnw -Dsurefire.useFile=false -DskipTests -Denforcer.skip=true clean install javadoc:aggregate --projects '!unit-tests,!serialization-tests,!jcstress-tests,!junit-trait-runner,!unit-tests-java8,!test-coverage,!p2-repository,!org.eclipse.collections:org.eclipse.collections'
+        env:
+          MAVEN_OPTS: "-Xmx1g"
+      - name: JavaDoc Jar
+        run: ./mvnw -Dsurefire.useFile=false -DskipTests -Denforcer.skip=true clean install javadoc:jar --projects '!unit-tests,!serialization-tests,!jcstress-tests,!junit-trait-runner,!unit-tests-java8,!test-coverage,!p2-repository,!org.eclipse.collections:org.eclipse.collections'
+        env:
+          MAVEN_OPTS: "-Xmx1g"
+
+  javadoc-early-access:
+    name: Javadoc Early Access
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download JDK 23 from jdk.java.net
+        uses: oracle-actions/setup-java@v1
+        id: download-jdk
+        with:
+          website: jdk.java.net
+          release: 23
+      - name: 'Print outputs'
+        shell: bash
+        run: |
+          echo 'Outputs'
+          echo "steps.download-jdk.outputs.archive = ${{ steps.download-jdk.outputs.archive }}"
+          echo "steps.download-jdk.outputs.version = ${{ steps.download-jdk.outputs.version }}"
+      - run: mvn --non-recursive wrapper:wrapper -Dmaven=3.9.6
+      - run: java --version
+      - name: Javadoc
+        run: ./mvnw -Dsurefire.useFile=false -DskipTests -Denforcer.skip=true clean install javadoc:aggregate -pl '!unit-tests,!serialization-tests,!jcstress-tests,!junit-trait-runner,!unit-tests-java8,!test-coverage,!p2-repository,!org.eclipse.collections:org.eclipse.collections'
```

Comment:
@motlin , you left this one as `-pl `instead of `--projects` as you have above in the javadoc job section.

Reply:
@donraab replaced both `-pl` with `--projects`